Public/Sdk/Public/Managed/Tools/ILCompiler/ilc.dsc

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import {Artifact, Cmd, Tool, Transformer} from "Sdk.Transformers";
+import * as Shared from "Sdk.Managed.Shared";
+
+const isMacOS = Context.getCurrentHost().os === "macOS";
+
+const pkgContents = isMacOS
+    ? importFrom("runtime.osx-x64.Microsoft.DotNet.ILCompiler").Contents.all
+    : importFrom("runtime.win-x64.Microsoft.DotNet.ILCompiler").Contents.all;
+
+const netcoreAppPkgContents = isMacOS
+    ? importFrom("runtime.osx-x64.Microsoft.NETCore.App").Contents.all
+    : importFrom("runtime.win-x64.Microsoft.NETCore.App").Contents.all;
+
+const ilcToolPackagePath = isMacOS
+    ? r`tools/ilc`
+    : r`tools/ilc.exe`;
+
+const ilcTool: Transformer.ToolDefinition = Shared.Factory.createTool({
+    exe: pkgContents.getFile(ilcToolPackagePath),
+    runtimeDependencies: pkgContents.contents,
+    dependsOnCurrentHostOSDirectories: true
+});
+
+@@public
+export const linkTimeLibraries: File[] =
+    pkgContents.getFiles(isMacOS ? [
+        r`sdk/libbootstrapper.a`,
+        r`sdk/libRuntime.a`,
+        r`sdk/libSystem.Private.CoreLib.Native.a`,
+        r`sdk/libSystem.Private.TypeLoader.Native.a`,
+        r`framework/System.Native.a`,
+        r`framework/System.Globalization.Native.a`,
+        r`framework/System.IO.Compression.Native.a`,
+        r`framework/System.Net.Http.Native.a`,
+        r`framework/System.Net.Security.Native.a`,
+        r`framework/System.Security.Cryptography.Native.Apple.a`
+    ] : [
+        // TODO: libraries for Windows
+    ]);
+
+@@public
+export const compileTimeReferences: File[] = [
+    ...pkgContents.contents.filter(f => f.name.extension === a`.dll` && f.path.parent.name !== a`tools`),
+    ...(isMacOS ? netcoreAppPkgContents.getFiles([
+        r`runtimes/osx-x64/lib/netcoreapp2.2/SOS.NETCore.dll`,
+        r`runtimes/osx-x64/lib/netcoreapp2.2/System.Runtime.InteropServices.WindowsRuntime.dll`
+    ]) : [
+        // TODO: references for Windows
+    ])
+];
+
+@@public
+export const defaultArgs = <Arguments>{
+    out: undefined,
+    inputs: undefined,
+    initAssemblies: [
+        "System.Private.CoreLib",
+        "System.Private.DeveloperExperience.Console",
+        "System.Private.StackTraceMetadata",
+        "System.Private.TypeLoader",
+        "System.Private.Reflection.Execution",
+        "System.Private.Interop"
+    ],
+    references: compileTimeReferences,
+    stackTraceData: true,
+    scanReflection: true,
+    emitDebugInformation: true,
+    completeTypeMetadata: true,
+    rootAllApplicationAssemblies: true,
+    dependencies: [
+        pkgContents
+    ]
+};
+
+@@public
+export interface Arguments extends Transformer.RunnerArguments {
```

Comment:
If there is an overview somewhere a comment with a link to what does options mean would be helpful.

Reply:
I copy pasted option descriptions from `ilc --help`